src/applications/personalization/profile/components/notification-settings/MissingContactInfoExpandable.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { FIELD_NAMES } from '@@vap-svc/constants';
+import { Link } from 'react-router-dom';
+import { VaAlertExpandable } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { NOTIFICATION_CHANNEL_IDS, PROFILE_PATHS } from '../../constants';
+import { useContactInfoDeepLink } from '../../hooks';
+import { useNotificationSettingsUtils } from '../../hooks/useNotificationSettingsUtils';
+
+const channelDescriptions = {
+  [NOTIFICATION_CHANNEL_IDS.EMAIL]: 'email address',
+  [NOTIFICATION_CHANNEL_IDS.TEXT]: 'mobile number',
+};
+
+const AddYourInfoDescription = ({ channel }) => {
+  const description = channelDescriptions[channel.id];
+
+  if (description) {
+    return (
+      <p className="vads-u-margin-top--0">
+        {`Add your ${description} to your profile to manage these ${
+          channel.name
+        } notification settings:`}
+      </p>
+    );
+  }
+
+  return null;
+};
+
+const UnavailableItemsList = ({ items }) => {
+  return (
+    <ul>
+      {items.map(item => (
+        <li key={item.name}>{item.name}</li>
+      ))}
+    </ul>
+  );
+};
+
+const AddYourInfoLink = ({ channel }) => {
+  const { generateContactInfoLink } = useContactInfoDeepLink();
+
+  const fieldNames = {
+    [NOTIFICATION_CHANNEL_IDS.EMAIL]: FIELD_NAMES.EMAIL,
+    [NOTIFICATION_CHANNEL_IDS.TEXT]: FIELD_NAMES.MOBILE_PHONE,
+  };
+
+  return (
+    <Link
+      to={generateContactInfoLink({
+        fieldName: fieldNames[channel.id],
+        focusOnEditButton: true,
+        returnPath: encodeURIComponent(PROFILE_PATHS.NOTIFICATION_SETTINGS),
+      })}
+    >
+      {`Add your ${channelDescriptions[channel.id]} to your profile`}
+    </Link>
+  );
+};
+
+export const MissingContactInfoExpandable = ({
+  showEmailNotificationSettings = false,
+}) => {
+  const {
+    useUnavailableItems,
+    missingChannels,
+  } = useNotificationSettingsUtils();
+
+  const unavailableItems = useUnavailableItems();
+
+  if (
+    !showEmailNotificationSettings ||
+    missingChannels.length > 1 ||
+    missingChannels.length === 0
```

Comment:
You know, now that I think about it, I think I could remove the `missingChannels.length > 1` part altogether, since that is covered within the outer component where it does a preliminary check to render this whole component or not.

Reply:
Nevermind that isn't true actually. I think `!== 1` is a good simplification.

Reply:
updated